### ISSUE TITLE
Cannot insert a record after querying record from a table (Azure v0.7.0)

### ISSUE THREAD

**Issue by AbelHu** [Azure/azure-sdk-for-ruby#281]
Always fail in inserting a record after querying records from a table.
The root cause is that the second requests uses the same Faraday connection with the first URI.

ARM returns below error:One of the request inputs is not valid.

The key is not correct in [http_client.rb] (https://github.com/Azure/azure-sdk-for-ruby/blob/master/service_management/azure/lib/azure/http_client.rb#L51-L59)
```
    def agents(uri)
      uri = URI.parse(uri) if uri.is_a?(String)
      key = uri.scheme.to_s + uri.host.to_s + uri.port.to_s
      @agents ||= {}
      unless @agents.key?(key)
        @agents[key] = build_http(uri)
      end
      @agents[key]
    end
```

**Comment by yaxia** [comment]
It is fixed in azure-storage 0.10.1. Please take try. Thanks.

**Comment by AbelHu** [comment]
azure-storage-ruby works. Thanks.
